### PR TITLE
Add Contravariant documentation page

### PR DESCRIPTION
**Pull request by vpavkin**
Preview http://vpavkin.github.io/cats/tut/contravariant.html

### CODE REVIEW

**Comment by vpavkin** [comment]
Appreciate any input/corrections :)

File: docs/src/main/tut/contravariant.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+layout: default
+title:  "Contravariant"
+section: "typeclasses"
+source: "https://github.com/non/cats/blob/master/core/src/main/scala/cats/functor/Contravariant.scala"
```
Comment (ceedubs) [inline docs/src/main/tut/contravariant.md]: This should switch from `non` to `typelevel`. We probably have the same mistake in lots of docs. We could probably just give relative paths here and use site-wide config for the base source URL.

File: docs/src/main/tut/contravariant.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+The `Contravariant` type class is for functors that define a `contramap`
+function with the following type:
+
+```scala
+def contramap[A, B](f: B => A): F[A] => F[B]
+```
+
+It looks like regular (also called `Covariant`) [`Functor`](functor.html)'s `map`,
+but with `f` transformation reversed.
```
Comment (ceedubs) [inline docs/src/main/tut/contravariant.md]: Might read more smoothly with as "but with _the_ `f` transformation..."

File: docs/src/main/tut/contravariant.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+Generally speaking, if you have some context `F[A]` for type `A`,
+and you can get `A` value out of `B` value — `Contravariant` allows you to get the `F[B]` context for `B`.
```
Comment (ceedubs) [inline docs/src/main/tut/contravariant.md]: maybe "you can get _an_ `A` value out of a `B` value"?

File: docs/src/main/tut/contravariant.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+Say we have class `Money` with a `Show` instance, and `Salary` class. 
+
+```tut:silent
+import cats._
+import cats.implicits._
+
+case class Money(amount: Int)
+case class Salary(size: Money)
+
+implicit val showMoney: Show[Money] = Show.show(m => s"${m.amount}$$")
```
Comment (ceedubs) [inline docs/src/main/tut/contravariant.md]: I'm used to the `$` being before the amount. Is this another weird thing that only the US does? :)

File: docs/src/main/tut/contravariant.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+If we want to show a `Salary` instance, we can just convert it to a `Money` instance and show it instead.
```
Comment (ceedubs) [inline docs/src/main/tut/contravariant.md]: I was fully expecting a Jerry Maguire quote to make it's way in here. Have I shown my age? :)

**Comment by ceedubs** [comment]
I made a few minor comments, but this is great. Thanks, @vpavkin!

File: docs/src/main/tut/contravariant.md (hunk 1 above)
Comment (vpavkin) [inline docs/src/main/tut/contravariant.md]: Strange, but currently `useful links` section, with those `source` links is hidden... 
I think this issue and all links should be fixed in another pr. I'll take care of that.

**Comment by vpavkin** [comment]
@ceedubs I'm not a native speaker, thank you for covering my back here! :)
Fixed the minors, will open a separate pr with sources links fixed

**Comment by ceedubs** [comment]
:+1:

**Comment by non** [comment]
Thanks so much! :+1:

**Pull request merged**